schematools/types.py

```diff
@@ -44,6 +44,27 @@ class DatasetSchema(SchemaType):
     This is a collection of JSON Schema's within a single file.
     """
 
+    _datasets_cache = {}
+
+    def __init__(
+        self, *args, datasets_cache: typing.Dict[str, DatasetSchema] = None, **kwargs
+    ):
+        """When initializing a datasets, a cache of related datasets
+        can be added (at classlevel). Thus, we are able to get (temporal) info
+        about the related datasets
+        """
+        super().__init__(*args, **kwargs)
+        if datasets_cache is not None:
+            self._datasets_cache = datasets_cache
```

Comment:
NIet `self._datasets_cache = (datasets_cache if datasets_cache is not None else {})` doen?
dan voorkom je dat het globale gedeelde object kan worden aangepast. Tenzij dat de bedoeling is natuurlijk.

Reply:
Ja, hier wil ik juist een globale cache op class niveau hebben, zodat je, ivm relaties, vanuit een dataset de andere datasets kunt "zien". Idealiter moet er een entiteit komen die alle datasets omvat (soort van dataset manager). Het zou een hoop refactoring vergen om die overal te introduceren, vandaar nu dit classlevel attribuut.